[DOCS] Revises doc on adding data (#68038)
elastic/kibana: docs/setup/connect-to-elasticsearch.asciidoc

[[connect-to-elasticsearch]]
== Adding data

To start working with your data in {kib}, you can:

* Use the File Data Visualizer or GeoJSON Upload feature to quickly load
data into {es}.

* Follow one of the Add Data tutorials to set up a Beats module to index logs,
metrics, events, or application data.

* Connect {kib} with existing {es} indices.

{kib} also provides <<get-data-in, sample data sets>> for you to use.

[float]
[[upload-data-kibana]]
=== Use the File Data Visualizer

> If your data is in a CSV,  JSON, or log file, you can

Reviewer: ```suggestion
To visualize data in a CSV,  JSON, or log file, you can
```

upload it using the File Data Visualizer.  On the home page,
click *Import a CSV, NDSON, or log file*, and then drag your file into the
File Data Visualizer. You can upload a file up to 100 MB.

[role="screenshot"]
image::images/add-data-fv.png[File Data Visualizer]

The File Data Visualizer uses the {ref}/ml-find-file-structure.html[find_file_structure API] to analyze
the uploaded file and to suggest ingest pipelines and mappings for your data.

NOTE: This feature is not intended for use as part of a
repeated production process, but rather for the initial exploration of your data.

[float]
[[upload-geoipdata-kibana]]
=== Use GeoJSON Upload

> If your data is a point or shape file, you can upload it using the <<geojson-upload, GeoJSON Upload>>

Reviewer: ```suggestion
To visualize geo data in a point or shape file, you can upload it using the <<geojson-upload, GeoJSON Upload>>
```

feature in *Elastic Maps*, and then use that data as a layer in a map.
The data is also available for use in the broader Kibana ecosystem, for example,
in visualizations and Canvas workpads.
With GeoJSON Upload, you can upload a file up to 50 MB.

[role="screenshot"]
image::images/add-data-geo-upload.png[GeoJSON Upload]

[float]
[[add-data-tutorial-kibana]]
===  Add data tutorials

The built-in data tutorials can help you quickly get up and running with
metrics data, log analytics, and security events.
These tutorials walk you through installing and configuring a

> Beats data shipper to periodically collect and send data to the Elastic Stack.

Reviewer: ```suggestion
Beats data shipper to periodically collect and send data to {es}.
```

You can then use the pre-built dashboards to explore and analyze the data.

You access the tutorials from the {kib} home page.
If a tutorial doesn’t exist for your data, go to the {beats-ref}/beats-reference.html[Beats overview]
to learn about other data shippers in the Beats family.

[role="screenshot"]
image::images/add-data-tutorials.png[Add Data tutorials]

[float]
[[connect-to-es]]
=== Connect with {es} indices

> If you want to connect with existing {es} indices, you must
> create an index pattern. {kib} requires an index pattern to identify
> which indices you want to explore. When you add data with the File Visualizer, GeoJSON Upload,

Reviewer: ```suggestion
To visualize data in existing {es} indices, you
create index patterns that match the names of the indices you want to explore. When you add data with the File Visualizer, GeoJSON Upload,
```

or an Add Data tutorial, an index pattern is created for you.

////
TIP: By default, {kib} connects to the {es} instance running on `localhost`. To connect to a
different {es} instance, modify the {es} URL in the `kibana.yml` configuration file and
restart {kib}. For information about using {kib} with your production nodes, see <<production>>.
////

. Go to *Stack Management* and then click *Index Patterns*.

. Click *Create index pattern*.

. Specify an index pattern that matches the name of one or more of your Elasticsearch indices.
+
For example, an index pattern can point to your Apache data from yesterday,
`filebeat-apache-4-3-2022`, or any index that matches the pattern, `filebeat-*`.
Using a wildcard is the more popular approach.

. Click *Next Step*, and then select the index field that contains the timestamp you want to use to perform time-based
comparisons.
+
Kibana reads the index mapping and lists all fields that contain a timestamp. If your
index doesn't have time-based data, choose *I don't want to use the Time Filter*.
+

> Without a default time field you will not be able to use global time filters on your dashboards.

Reviewer: ```suggestion
You must select a time field to use global time filters on your dashboards.
```

. Click *Create index pattern*.
+

> {kib} is now configured to use your Elasticsearch data.

Reviewer: ```suggestion
{kib} is now configured to access your {es} indices.
```

You’ll see a list of fields configured for the matching index.
You can designate your index pattern the default by clicking the star icon on this page.
+

> When you search your data in Discover or create a visualization, select your index pattern from
> the index pattern menu.

Reviewer: ```suggestion
When searching in Discover and creating visualizations, you choose a pattern from the index pattern menu to specify the {es} indices that contain the data you want to work with.
```

[[connect-to-elasticsearch]]

> == Adding data

Reviewer 2: ```suggestion
== Add data
```

Author: I'll make that change when I change the other titles in the section.

To start working with your data in {kib}, you can:

> * Use the File Data Visualizer or GeoJSON Upload feature to quickly load

Reviewer 2: ```suggestion
* Quickly load data into Elasticsearch with the File Data Visualizer or GeoJSON Upload feature
```

> * Use the File Data Visualizer or GeoJSON Upload feature to quickly load

Reviewer 2: Do we really want to say `File Data Visualizer or GeoJSON Upload feature` before we tell them what these features are? I would pair down this list to:

- Load your data from a log file
- Index logs, metrics, events, and application data
- Connect Kibana to your Elasticsearch indices

Then get into the nitty gritty in the sections below.

Reviewer 3: > Load your data from a log file

It might be worth mentioning CSV files here also as they are a very common use case.

data into {es}.

* Follow one of the Add Data tutorials to set up a Beats module to index logs,
metrics, events, or application data.

* Connect {kib} with existing {es} indices.

> {kib} also provides <<get-data-in, sample data sets>> for you to use.

Reviewer 2: How about:

If you're not ready to use your own data, use the <<get-data-in, sample data sets>> to see all that you can do in {kib}.

[float]
[[upload-data-kibana]]

> === Use the File Data Visualizer

Reviewer 2: How about:

Load your data from a log file

To visualize data in a CSV,  JSON, or log file, you can
upload it using the File Data Visualizer.  On the home page,
click *Import a CSV, NDSON, or log file*, and then drag your file into the

> File Data Visualizer. You can upload a file up to 100 MB.

Reviewer 3: This value is configurable up to 1GB in kibana's advanced settings. It might be worth mentioning in these docs.

[role="screenshot"]
image::images/add-data-fv.png[File Data Visualizer]

The File Data Visualizer uses the {ref}/ml-find-file-structure.html[find_file_structure API] to analyze
the uploaded file and to suggest ingest pipelines and mappings for your data.

> NOTE: This feature is not intended for use as part of a

Reviewer 2: I'm not sure what this note means. Why wouldn't I be able to use this feature repeatedly? 

The second part, `rather for the initial exploration of your data` sounds like part of the value prop.

repeated production process, but rather for the initial exploration of your data.

[float]
[[upload-geoipdata-kibana]]

> === Use GeoJSON Upload

Reviewer 2: How about:

Load your GeoJSON data

> === Use GeoJSON Upload

Reviewer 2: I see `load` and `upload`. Maybe choose one?

> To visualize geo data in a point or shape file, you can upload it using the <<geojson-upload, GeoJSON Upload>>

Reviewer 2: At this point, do users know what point or shape files are?

> feature in *Elastic Maps*, and then use that data as a layer in a map.

Reviewer 2: *Elastic Maps* now appears as *Maps* in the new nav structure

> feature in *Elastic Maps*, and then use that data as a layer in a map.

Reviewer 2: If users have to go into Maps to use this feature, then does it really belong here? Or does it belong in the Maps section? The other options are available from the home page, right?

The data is also available for use in the broader Kibana ecosystem, for example,
in visualizations and Canvas workpads.
With GeoJSON Upload, you can upload a file up to 50 MB.

[role="screenshot"]
image::images/add-data-geo-upload.png[GeoJSON Upload]

[float]
[[add-data-tutorial-kibana]]

> ===  Add data tutorials

Reviewer 2: Your'e not really adding data tutorials, you're using them.

The built-in data tutorials can help you quickly get up and running with
metrics data, log analytics, and security events.
These tutorials walk you through installing and configuring a
Beats data shipper to periodically collect and send data to {es}.
You can then use the pre-built dashboards to explore and analyze the data.

> You access the tutorials from the {kib} home page.

Reviewer 2: ```suggestion
You access the tutorials from the home page.
```

If a tutorial doesn’t exist for your data, go to the {beats-ref}/beats-reference.html[Beats overview]
to learn about other data shippers in the Beats family.

[role="screenshot"]
image::images/add-data-tutorials.png[Add Data tutorials]

[float]
[[connect-to-es]]
=== Connect with {es} indices

> To visualize data in existing {es} indices, you

Reviewer 2: ```suggestion
To visualize data in {es} indices, you
```

> create index patterns that match the names of the indices you want to explore. When you add data with the File Visualizer, GeoJSON Upload,

Reviewer 2: ```suggestion
create index patterns that match the names of the indices you want to explore. When you add data with the File Data Visualizer, GeoJSON Upload,
```

or an Add Data tutorial, an index pattern is created for you.

> 

Reviewer 2: To create an index pattern:
